src/components/Settings/MaxSlippageSettings/index.tsx

```diff
@@ -0,0 +1,174 @@
+  const tooLow =
+    userSlippageTolerance !== SlippageTolerance.Auto && userSlippageTolerance.lessThan(MINIMUM_RECOMMENDED_SLIPPAGE)
+  const tooHigh =
+    userSlippageTolerance !== SlippageTolerance.Auto && userSlippageTolerance.greaterThan(MAXIMUM_RECOMMENDED_SLIPPAGE)
+
+  return (
+    <Expand
+      header={
+        <Row width="auto">
+          <ThemedText.BodySecondary>
+            <Trans>Max slippage</Trans>
+          </ThemedText.BodySecondary>
+          <QuestionHelper
+            text={
+              <Trans>Your transaction will revert if the price changes unfavorably by more than this percentage.</Trans>
+            }
+          />
+        </Row>
+      }
+      button={
+        <ThemedText.BodyPrimary>
+          {userSlippageTolerance === SlippageTolerance.Auto ? (
+            <Trans>Auto</Trans>
+          ) : (
+            `${userSlippageTolerance.toFixed(2)}%`
+          )}
+        </ThemedText.BodyPrimary>
+      }
+    >
+      <RowBetween gap="md">
+        <Switch>
+          <Option
+            onClick={() => {
+              // Reset the input field when switching to auto
+              setSlippageInput('')
+              setUserSlippageTolerance(SlippageTolerance.Auto)
+            }}
+            isActive={userSlippageTolerance === SlippageTolerance.Auto}
+          >
+            <ThemedText.BodyPrimary>
+              <Trans>Auto</Trans>
+            </ThemedText.BodyPrimary>
+          </Option>
+          <Option
+            onClick={() => {
+              // When switching to custom slippage, use `auto` value as a default.
+              setUserSlippageTolerance(autoSlippage)
+            }}
+            isActive={userSlippageTolerance !== SlippageTolerance.Auto}
+          >
+            <ThemedText.BodyPrimary>
+              <Trans>Custom</Trans>
+            </ThemedText.BodyPrimary>
+          </Option>
+        </Switch>
+        <InputContainer gap="md" error={!!slippageError}>
+          <Input
+            data-testid="slippage-input"
+            placeholder={autoSlippage.toFixed(2)}
+            value={slippageInput}
+            onChange={(e) => parseSlippageInput(e.target.value)}
+            onBlur={() => {
+              // When the input field is blurred, reset the input field to the default value
+              setSlippageInput(defaultSlippageInputValue)
+              setSlippageError(false)
+            }}
+          />
+          <ThemedText.BodyPrimary>%</ThemedText.BodyPrimary>
+        </InputContainer>
+      </RowBetween>
+      {tooLow || tooHigh ? (
+        <RowBetween gap="md">
+          <CautionTriangle />
+          <ThemedText.Caption color="accentWarning">
+            {tooLow ? (
+              <Trans>
+                Slippage below {MINIMUM_RECOMMENDED_SLIPPAGE.toFixed(2)}% may result in a failed transaction
+              </Trans>
+            ) : (
+              <Trans>Your transaction may be frontrun and result in an unfavorable trade.</Trans>
+            )}
```

Comment:
I think this would be clearer if you guarded: `{ tooLow && <...` so that you could see both values.

Reply:
It is not possible for both values to be present at the same time, since one value can't be both too low and too large. That said, I updated the code to your suggested pattern and added a test case to guard against two showing up at the same time. It's a left-over from the previous code.